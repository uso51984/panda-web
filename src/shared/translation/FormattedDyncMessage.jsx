import React from 'react';
import decodeHtmlStr from 'shared/utils/decodeHtmlStr';

const FormattedDyncMessage = ({ value }) => (
  <>
    {decodeHtmlStr(`${value || ''}`)}
  </>
);

export default FormattedDyncMessage;
