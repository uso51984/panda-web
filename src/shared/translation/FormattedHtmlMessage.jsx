import React from 'react';
import decodeHtmlStr from '@/shared/utils/decodeHtmlStr';

/**
 * TODO: prevent XSS.
 */

const FormattedHtmlMessage = ({ value, ...rest }) => (
  // eslint-disable-next-line react/no-danger
  <span {...rest} dangerouslySetInnerHTML={{ __html: decodeHtmlStr(`${value || ''}`) }} />
);

export default FormattedHtmlMessage;
