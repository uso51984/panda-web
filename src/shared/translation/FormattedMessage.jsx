import React from 'react';
import * as ReactIntl from 'react-intl';
import decodeHtmlStr from 'shared/utils/decodeHtmlStr';

const FormattedMessage = (props) => {
  const { children, values, ...rest } = props;
  const formattedValues = {};
  if (values) {
    Object.keys(values).forEach((key) => {
      const val = values[key];
      formattedValues[key] = typeof val === 'string' ? decodeHtmlStr(val) : val;
    });
  }
  return (
    <ReactIntl.FormattedMessage {...rest} values={formattedValues}>
      {children}
    </ReactIntl.FormattedMessage>
  );
};

export default FormattedMessage;
