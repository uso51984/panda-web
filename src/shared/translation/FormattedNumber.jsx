import React from 'react';
import { formattedNumberFunc } from './utils';

const FormattedNumber = ({ value, numberStyle }) => (
  <span className="format-number">
    {formattedNumberFunc({ value, numberStyle })}
  </span>
);

export default FormattedNumber;
