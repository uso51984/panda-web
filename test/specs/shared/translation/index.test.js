import React from 'react';
import renderer from 'react-test-renderer';
import FormattedDyncMessage from '@/shared/translation/FormattedDyncMessage';
import FormattedHtmlMessage from '@/shared/translation/FormattedHtmlMessage';
import FormattedMessage from '@/shared/translation/FormattedMessage';
import FormattedNumber from '@/shared/translation/FormattedNumber';

describe('translation', () => {
  it('FormattedDyncMessage should work fine ', () => {
    const tree = renderer
      .create(<div><FormattedDyncMessage value="test" /></div>)
      .toJSON();
    expect(tree).toMatchSnapshot();

    const tree1 = renderer
      .create(<div><FormattedDyncMessage value="" /></div>)
      .toJSON();
    expect(tree1).toMatchSnapshot();

  });


  it('FormattedHtmlMessage should work fine ', () => {
    const tree = renderer
      .create(<FormattedHtmlMessage value="FormattedHtmlMessage" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('FormattedNumber should work fine ', () => {
    const tree = renderer
      .create(<FormattedNumber value={233} numberStyle="decimalinreport"  />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
