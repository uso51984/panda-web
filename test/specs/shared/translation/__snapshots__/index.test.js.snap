// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translation FormattedDyncMessage should work fine  1`] = `
<div>
  test
</div>
`;

exports[`translation FormattedDyncMessage should work fine  2`] = `
<div>
  
</div>
`;

exports[`translation FormattedHtmlMessage should work fine  1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "FormattedHtmlMessage",
    }
  }
/>
`;

exports[`translation FormattedNumber should work fine  1`] = `
<span
  className="format-number"
>
  $233.00
</span>
`;
